signer/src/storage/postgres.rs

```diff
@@ -218,38 +218,102 @@ impl super::DbRead for PgStore {
     async fn get_bitcoin_canonical_chain_tip(
         &self,
     ) -> Result<Option<model::BitcoinBlockHash>, Self::Error> {
-        todo!(); // TODO(244): Write query + integration test
+        sqlx::query_as!(
+            model::BitcoinBlock,
+            "SELECT * FROM sbtc_signer.bitcoin_blocks ORDER BY block_height DESC, block_hash DESC"
+        )
+        .fetch_optional(&self.0)
+        .await
+        .map(|maybe_block| maybe_block.map(|block| block.block_hash))
+        .map_err(Error::SqlxQuery)
     }
 
     async fn get_pending_deposit_requests(
         &self,
-        _chain_tip: &model::BitcoinBlockHash,
-        _context_window: usize,
+        chain_tip: &model::BitcoinBlockHash,
+        context_window: i32,
     ) -> Result<Vec<model::DepositRequest>, Self::Error> {
-        todo!(); // TODO(244): Write query + integration test
+        sqlx::query_as!(
+            model::DepositRequest,
+            r#"
+            WITH RECURSIVE ContextWindow AS (
+                -- Anchor member: Initialize the recursion with the chain tip
+                SELECT block_hash, block_height, parent_hash, created_at, 1 AS depth
+                FROM sbtc_signer.bitcoin_blocks
+                WHERE block_hash = $1
+                
+                UNION ALL
+                
+                -- Recursive member: Fetch the parent block using the last block's parent_hash
+                SELECT parent.block_hash, parent.block_height, parent.parent_hash,
+                       parent.created_at, last.depth + 1
+                FROM sbtc_signer.bitcoin_blocks parent
+                JOIN ContextWindow last ON parent.block_hash = last.parent_hash
+                WHERE last.depth < $2
+            ),
+            TransactionsInWindow AS (
+                SELECT transactions.txid
+                FROM ContextWindow blocks_in_window
+                JOIN sbtc_signer.bitcoin_transactions transactions ON
+                    transactions.block_hash = blocks_in_window.block_hash
+            ),
+            DepositsInWindow AS (
+                SELECT deposit_requests.*
+                FROM TransactionsInWindow transactions
+                JOIN sbtc_signer.deposit_requests deposit_requests ON
+                    deposit_requests.txid = transactions.txid
+            )
+            SELECT * FROM DepositsInWindow;
```

Comment:
Why not just
```sql
),
SELECT deposit_requests.*
FROM TransactionsInWindow transactions
JOIN sbtc_signer.deposit_requests deposit_requests ON
    deposit_requests.txid = transactions.txid;
```

Reply:
Right, that's probably a working simplification. I'll try it out. Thanks for suggesting! 🙏

Reply:
Done.